### PR TITLE
Add onClick prop to NavLink

### LINKED ISSUE #4491: Dropdown nav menu doesn't close on phone screens
## Expected behavior
After navigating to a new page, the menu should close.

## Current behavior
_Please include any error messages from the browser console and/or screenshots_
After navigating from Talk to Classify, the menu stays open, obscuring the subject.
screenshot from Snapshot Serengeti, showing the nav menu in front of the subject photo.

## Steps to replicate
Open the menu from a project Talk, on your phone, and select Classify.

## Additional information

- **Operating system:** iOS 11
- **Browser:** Safari

### PR DESCRIPTION
**Pull request by eatyourgreens**
https://fix-4491.pfe-preview.zooniverse.org/

Pass the custom click handler down to nav links.
Remove the unused site prop.

Fixes #4491

# Required Manual Testing

- [ ] Does the non-logged in home page render correctly?
- [ ] Does the logged in home page render correctly?
- [ ] Does the projects page render correctly?
- [ ] Can you load project home pages?
- [ ] Can you load the classification page?
- [ ] Can you submit a classification?
- [ ] Does talk load correctly?
- [ ] Can you post a talk comment?

# Review Checklist

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [ ] If the component is in coffeescript, is it converted to ES6? Is it free of eslint errors? Is the conversion its own commit?
- [ ] Are the tests passing locally and on Travis?

# Optional

- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you [resized and compressed](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/image-optimization) any image you've added?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?
- [ ] Have you followed the [Springer guidelines for commit messages](https://github.com/springernature/frontend-playbook/blob/master/git/git.md#commit-messages)?

### CODE REVIEW

File: app/pages/project/components/ProjectNavbar/components/NavLink/NavLink.jsx [hunk 1]
```diff
@@ -70,7 +71,7 @@ NavLink.defaultProps = {
   disabled: false,
   isExternalLink: false,
   label: '',
-  site: '',
+  onClick: () => false,
```
Comment (eatyourgreens) [inline app/pages/project/components/ProjectNavbar/components/NavLink/NavLink.jsx]: I think this should actually return `true`.

Comment (eatyourgreens) [reply app/pages/project/components/ProjectNavbar/components/NavLink/NavLink.jsx]: https://stackoverflow.com/questions/18971284/event-preventdefault-vs-return-false-no-jquery

Comment (simoneduca) [reply app/pages/project/components/ProjectNavbar/components/NavLink/NavLink.jsx]: It doesn't look you'd want to cancel the event here, so I think you're right.

Comment (eatyourgreens) [reply app/pages/project/components/ProjectNavbar/components/NavLink/NavLink.jsx]: Yeah, with my original code, the default action is to cancel a click event, unless an `onClick` handler is explicitly given.

**Pull request merged**